### PR TITLE
test(NODE-3011): lb support spec test changes

### PR DESCRIPTION
**Pull request by durran**
Includes only the spec test changes from LB support.

### CODE REVIEW

File: test/functional/unified-spec-runner/operations.ts [hunk 1]
```diff
@@ -180,7 +180,8 @@ operations.set('assertNumberConnectionsCheckedOut', async ({ entities, operation
     const pool = server.s.pool;
     return count + pool.currentCheckedOutCount;
   }, 0);
-  expect(checkedOutConnections).to.equal(operation.arguments.connections);
+  // TODO: Durran: Fix in NODE-3011
```
Comment (durran) [inline test/functional/unified-spec-runner/operations.ts:183]: The count is implemented in the connection pool in NODE-3011 PR.

File: test/unit/sdam/server_selection/spec.test.js [hunk 2]
```diff
@@ -68,7 +68,9 @@ describe('Server Selection (spec)', function () {
         describe(subType, function () {
           specTests[topologyType][subType].forEach(test => {
             // NOTE: node does not support PossiblePrimary
-            const maybeIt = test.name.match(/Possible/) ? it.skip : it;
+            // TODO: Re-enable LoadBalanced in NODE-3011
+            const maybeIt =
+              test.name.match(/Possible/) || topologyType === 'LoadBalanced' ? it.skip : it;
```
Comment (durran) [inline test/unit/sdam/server_selection/spec.test.js:73]: Will re-enable these as part of NODE-3011

File: test/unit/core/mongodb_srv.test.js [hunk 3]
```diff
@@ -24,7 +24,9 @@ describe('mongodb+srv', function () {
         test[1].comment = test[0];
       }
 
-      it(test[1].comment, {
+      // TODO: Remove with NODE-3011
+      const maybeIt = test[1].comment.includes('loadBalanced') ? it.skip : it.only;
```
Comment (dariakp) [inline test/unit/core/mongodb_srv.test.js:28]: is .only actually what we want here? not just plain `it`?

**Review by dariakp — CHANGES_REQUESTED** [review]
Just noticed one thing, let me know if that's intentional

File: test/unit/core/mongodb_srv.test.js (hunk 3 above)
Comment (durran) [reply test/unit/core/mongodb_srv.test.js:28]: Whoops, that was for me locally. I'll update.

Comment (durran) [reply test/unit/core/mongodb_srv.test.js]: Updated.

**Review by dariakp — APPROVED** [review]

**Review by nbbeeken — APPROVED** [review]

**Review by emadum — APPROVED** [review]

**Pull request merged**